Lib/test/test_pstats.py

```diff
@@ -30,12 +33,35 @@ class StatsTestCase(unittest.TestCase):
     def setUp(self):
         stats_file = support.findfile('pstats.pck')
         self.stats = pstats.Stats(stats_file)
+        to_compile = 'import os'
+        self.temp_storage = tempfile.mktemp()
+        profiled = compile(to_compile, '<string>', 'exec')
+        cProfile.run(profiled, filename=self.temp_storage)
+
+    def tearDown(self):
+        remove(self.temp_storage)
 
     def test_add(self):
         stream = StringIO()
         stats = pstats.Stats(stream=stream)
         stats.add(self.stats, self.stats)
 
+    def test_dump_and_load_works_correctly(self):
+        self.stats.dump_stats(filename=self.temp_storage)
+        tmp_stats = pstats.Stats(self.temp_storage)
+        self.assertEqual(self.stats.stats, tmp_stats.stats)
+
+    def test_load_equivalent_to_init(self):
+        empty = pstats.Stats()
+        empty.load_stats(self.temp_storage)
```

Comment:
It seems like your new tests both dump to and load from this temp file, functionality wise it works for now, but I think we should "isolate" the tests as much as possible? Reordering the execution of the tests could cause this `load_stats` to load different files - it could be the file created at `setUp`, or it could be the file created in `test_dump_and_load_works_correctly`. Again, it works because the test does not care what's in the file, as long as it's valid, it's just a bit convoluted. It seems like the profile created in `setUp` is only (potentially) used in this test? Maybe consider to move that piece into the unittest directly? The other test will overwrite thie file anyway I assume?

Reply:
addressed in https://github.com/python/cpython/pull/111447/files